File: languages/java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/AppEncryptionBytesImpl.java
package com.godaddy.asherah.appencryption;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.godaddy.asherah.appencryption.envelope.EnvelopeEncryption;

public class AppEncryptionBytesImpl<D> implements AppEncryption<byte[], D> {
  private static final Logger logger = LoggerFactory.getLogger(AppEncryptionBytesImpl.class);

  private final EnvelopeEncryption<D> envelopeEncryption;

  public AppEncryptionBytesImpl(final EnvelopeEncryption<D> envelopeEncryption) {
    this.envelopeEncryption = envelopeEncryption;
  }

  @Override
  public byte[] decrypt(final D dataRowRecord) {
    return envelopeEncryption.decryptDataRowRecord(dataRowRecord);
  }

  @Override
  public D encrypt(final byte[] payload) {
    return envelopeEncryption.encryptPayload(payload);
  }

  @Override
  public void close() {
    try {
      envelopeEncryption.close();
    }
    catch (Exception e) {
      logger.error("unexpected exception during close", e);
    }
  }
}

File: languages/csharp/AppEncryption/AppEncryption/AppEncryptionBytesImpl.cs
using System;
using GoDaddy.Asherah.AppEncryption.Envelope;
using GoDaddy.Asherah.Logging;
using Microsoft.Extensions.Logging;

namespace GoDaddy.Asherah.AppEncryption
{
    public class AppEncryptionBytesImpl<TD> : AppEncryption<byte[], TD>
    {
        private static readonly ILogger Logger = LogManager.CreateLogger<AppEncryptionBytesImpl<TD>>();

        private readonly IEnvelopeEncryption<TD> envelopeEncryption;

        public AppEncryptionBytesImpl(IEnvelopeEncryption<TD> envelopeEncryption)
        {
            this.envelopeEncryption = envelopeEncryption;
        }

        public override byte[] Decrypt(TD dataRowRecord)
        {
            return envelopeEncryption.DecryptDataRowRecord(dataRowRecord);
        }

        public override TD Encrypt(byte[] payload)
        {
            return envelopeEncryption.EncryptPayload(payload);
        }

        public override void Dispose()
        {
            try
            {
                envelopeEncryption.Dispose();
            }
            catch (Exception e)
            {
                Logger.LogError(e, "unexpected exception during close");
            }
        }
    }
}
